core/src/main/scala/org/apache/spark/scheduler/ResultTask.scala

```diff
@@ -55,8 +56,17 @@ private[spark] class ResultTask[T, U](
     // Deserialize the RDD and the func using the broadcast variables.
     val deserializeStartTime = System.currentTimeMillis()
     val ser = SparkEnv.get.closureSerializer.newInstance()
+    // Kryo deserialization is not thread-safe w.r.t. underlying buffer
```

Comment:
I think you had a nice mailing list reference for this.  Maybe put a comment with the link here?  I'm always a fan of overly-verbose comments.